### PR TITLE
RemoteSpawnRunner: record inbetween phases in timing profile

### PR DESCRIPTION
**Pull request by sluongng**
After an action was executed remotely, RemoteSpawnRunner would use
the timestamps in the execution metadata to record appropriate timing
phases into the JSON profile.

However, there are durations in-between the existing phases that are
unaccounted for. Depending on the RBE server implemenation, these
phases could mean different things:
- Sandbox preparation
- Cleaning up sandbox environments post-execution
- Others

Missing these durations inside the timing profile would cause confusion
to end users as it would be interpreted as nothing happened in between
the existing phases.

Add these durations into the profile as "pre-X" phases so that user is
aware of activities could still be happening during that time. RBE
server implementation should be able to alter these label
programmatically if necessary.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java [hunk 1]
```diff
@@ -339,18 +339,36 @@ private static void profileAccounting(ExecutedActionMetadata executedActionMetad
...
         converter,
         executedActionMetadata.getInputFetchStartTimestamp(),
         executedActionMetadata.getInputFetchCompletedTimestamp(),
         REMOTE_SETUP,
         "fetch");
+    logProfileTask(
+        converter,
+        executedActionMetadata.getInputFetchCompletedTimestamp(),
+        executedActionMetadata.getExecutionStartTimestamp(),
+        REMOTE_QUEUE,
```
Comment (tjgq) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java:358]: I'm not sure how much it matters, but should this be `REMOTE_SETUP` to keep with the pattern?

**Review by coeuvre — APPROVED** [review]

**Comment by brentleyjones** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
@bazel-io fork 7.1.0

**Comment by iancha1992** [comment]
@bazel-io fork 6.5.0

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 6.5.0 RC1](https://releases.bazel.build/6.5.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.1.0 RC1](https://releases.bazel.build/7.1.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**